File: app/src/main/java/com/kazakago/cleanarchitecture/domain/repository/WeatherRepository.java
package com.kazakago.cleanarchitecture.domain.repository;

import com.kazakago.cleanarchitecture.domain.model.weather.WeatherModel;

import io.reactivex.Single;

/**
 * Weather Repository
 *
 * @author Kensuke
 */
public interface WeatherRepository {

    Single<WeatherModel> fetch(String cityId);

    WeatherModel find(String cityId);

}

File: app/src/main/java/com/kazakago/cleanarchitecture/domain/repository/WeatherRepository.kt
package com.kazakago.cleanarchitecture.domain.repository

import com.kazakago.cleanarchitecture.domain.model.weather.WeatherModel

interface WeatherRepository {

    fun find(cityId: String): WeatherModel?

    fun exist(cityId: String): Boolean

    fun insert(weather: WeatherModel)

    fun delete(weather: WeatherModel)

}
